dist2src/core.py

```diff
@@ -573,6 +590,21 @@ def convert_single_commit(self, origin_branch: str, dest_branch: str):
         # mark the last upstream commit
         self.source_git.create_tag(tag=source_git_tag, branch=dest_branch)
 
+    def set_sources_in_packit_config(self, config: dict):
+        config.setdefault("sources", [])
+        for sha, path in self.lookaside_sources():
+
+            # get_sources.sh does git magic in the repo to determine the correct branch
+            for branch in ("c8s", "c8"):
+                url = (
+                    f"https://git.centos.org/sources/{self.package_name}/{branch}/{sha}"
+                )
+                response = requests.head(url)
+                if not response.ok:
+                    continue
```

Comment:
In both places where `add_packit_config` is called, there is an `origin_branch` variable available to store which branch is being converted. Couldn't this be passed down to this function and used, instead of this probing mechanism?

Reply:
sadly, not; there are cases where the sources are stored on a different branch; example: we are processing c8s but the sources are on the c8 branch - there is some nasty git-magic in get_sources.sh to figure out the correct url

Reply:
Do you have an example for such a case? I wasn't able to find one. Had a look at [rsync](https://git.centos.org/sources/rsync/), but there I see that the same sources (same hash) show up in all the branches between `c8-beta` and `c8s`, which tells me that the lookaside cache does follow whatever branching is happening in Git.

The "magic" `get_sources.sh` does, is not "magic" at all: it just looks up all the branches the current HEAD belongs to, and tries to download the sources using all these branch names, until one of them succeeds. Which kinda is  similar to this implementation.

But, this mechanism [was introduced](https://git.centos.org/centos-git-common/c/c61c1ca3917f2377553df26a5290e5dfb8780631) to `get_sources.sh` in order to handle "detached branches" (meant to be "detached HEAD", I assume). When on a detached HEAD, it does makes sense to follow this approach. But dist2src is expected to always work on the tip of a branch, in which case I don't see why this guessing would be required.

Reply:
hmmm, apparently I understood the get_sources.sh code wrong (or things changed with the lookaside in the past couple of months - I clearly remember how someone from the CentOS team was explaining this behaviour to me last year)

I'll drop the magic and see what our test suite will say

thanks for the explanation